caffe2/python/examples/cifar10_fp16.py

```diff
@@ -0,0 +1,210 @@
+import numpy as np
+import os
+import time
+from timeit import default_timer as timer
+
+from caffe2.python import brew, core, cnn, net_drawer, workspace
+from caffe2.proto import caffe2_pb2
+from caffe2.python.model_helper import ModelHelper
+
+def AddInput(model, batch_size, db, db_type, param_type="float", is_test=False):
+    """Adds the data input part."""
+    reader = model.CreateDB(
+        "reader_"+db,
+        db=db,
+        db_type=db_type)
+
+    # Note: Using C2 datum for this test
+    data, label = model.ImageInput(
+        [reader], ["data", "label"],
+        use_gpu_transform=True,
+        batch_size=batch_size, use_caffe_datum=True,
+        mean=128., scale=32, crop=32, mirror=1, color=3,
+        is_test=is_test, output_type=param_type
+    )
+    data = model.StopGradient(data, data)
+
+    return data, label
+
+# Move conv1 to fp16 only
+def AddCifar10Model(model, data, param_type):
+
+    with brew.arg_scope([brew.conv, brew.fc], dtype="float16"):
+        conv1 = brew.conv(model, data, 'conv1', 3, 32, 5,
```

Comment:
What is the line lint rule? I don't see any utilities supplied with the code for me to lint myself...